app/screens/onboarding/index.tsx

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useRef} from 'react';
+import {
+    View,
+    useWindowDimensions,
+    SafeAreaView,
+    ScrollView,
+    StyleSheet,
+    Platform,
+    NativeSyntheticEvent,
+    NativeScrollEvent,
+} from 'react-native';
+import {Navigation} from 'react-native-navigation';
+import Animated, {useAnimatedStyle, useDerivedValue, useSharedValue, withTiming} from 'react-native-reanimated';
+
+import {storeOnboardingViewedValue} from '@actions/app/global';
+import {Screens} from '@constants';
+import Background from '@screens/background';
+import {goToScreen, loginAnimationOptions} from '@screens/navigation';
+
+import FooterButtons from './footer_buttons';
+import Paginator from './paginator';
+import SlideItem from './slide';
+import useSlidesData from './slides_data';
+
+import type {LaunchProps} from '@typings/launch';
+
+interface OnboardingProps extends LaunchProps {
+    theme: Theme;
+}
+
+const styles = StyleSheet.create({
+    onBoardingContainer: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+        verticalAlign: 'top',
+    },
+    scrollContainer: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+});
+
+const AnimatedSafeArea = Animated.createAnimatedComponent(SafeAreaView);
+
+const Onboarding = ({
+    theme,
+}: OnboardingProps) => {
+    const {width} = useWindowDimensions();
+    const {slidesData} = useSlidesData();
+    const LAST_SLIDE_INDEX = slidesData.length - 1;
+    const slidesRef = useRef<ScrollView>(null);
+
+    const scrollX = useSharedValue(0);
+
+    // used to smothly animate the whole onboarding screen during the appear event scenario (from server screen back to onboarding screen)
+    const translateX = useSharedValue(width);
+
+    const currentIndex = useDerivedValue(() => Math.round(scrollX.value / width));
+
+    const moveToSlide = useCallback((slideIndexToMove: number) => {
+        slidesRef.current?.scrollTo({
+            x: (slideIndexToMove * width),
+            animated: true,
+        });
+    }, [width]);
+
+    const signInHandler = useCallback(() => {
+        // mark the onboarding as already viewed
+        storeOnboardingViewedValue();
+
+        goToScreen(Screens.SERVER, '', {animated: true, theme}, loginAnimationOptions());
+    }, []);
+
+    const nextSlide = useCallback(() => {
+        const nextSlideIndex = currentIndex.value + 1;
+        if (slidesRef.current && currentIndex.value < LAST_SLIDE_INDEX) {
+            moveToSlide(nextSlideIndex);
+        } else if (slidesRef.current && currentIndex.value === LAST_SLIDE_INDEX) {
+            signInHandler();
+        }
+    }, [currentIndex, slidesRef.current, moveToSlide, signInHandler]);
```

Comment:
In theory, you shouldn't need slidesRef.current in the dependencies because it is a reference. You may need a dependency on slidesRef if you expect the reference to change (99% of the cases it doesn't), but probably you can remove it completely from the dependencies.